### PR TITLE
IOS-2662: Create separate targets for helpers using XCTest

### PR DESCRIPTION
**Pull request by lquigley**
**Issue Link**
IOS-2662

**Description**
Because SHAM and UtilityBeltUITesting will now be imported into the main SpotHero target via SpotHeroData, building errors are occurring when attempting to build the app. The underlying issue being that XCTest can be imported into the main target, but XCTest should not be present in the final archive.

These two extensions referencing XCTest have been separated into their own targets. They will be imported into the testing targets that need them in the main SpotHero workspace.

### CODE REVIEW

File: Sources/ShamTestHelpers/Extensions/XCTestCase+Stub.swift [hunk 1]
```diff
@@ -1,24 +1,25 @@
 // Copyright © 2020 SpotHero, Inc. All rights reserved.
 
-#if !os(watchOS) && canImport(XCTest)
-    
+#if canImport(XCTest)
```
Comment (bdrelling) [inline Sources/ShamTestHelpers/Extensions/XCTestCase+Stub.swift:3]: It's odd to me that doing this wouldn't work. I wonder if there's a better way besides splitting the modules up?

I don't think this PR is a bad solution at all, I just think if we could solve for a conditional statement for including this extension, it simplifies everything downstream as well.

Comment (bdrelling) [reply Sources/ShamTestHelpers/Extensions/XCTestCase+Stub.swift:3]: Like, maybe just doing the equivalent of a `os(x386)` or whatever the code would be, would mean this is only imported if being built for simulators and running `XCTest`?

Just spitballing.

Comment (lquigley) [reply Sources/ShamTestHelpers/Extensions/XCTestCase+Stub.swift:3]: @bdrelling Even if we limit it to just simulators, `targetEnvironment(simulator)`, building the main SpotHero project in Debug mode on a simulator will have errors - "Undefined symbol: __swift_FORCE_LOAD_$_XCTestSwiftSupport". I can get by these errors by setting "Enable Testing Search Paths" to "YES" in Build Settings on both SpotHeroData and SpotHero.

Another option is to set a specific flag on the test targets that validate on the if statement in the specific extensions, but I felt that it was cleaner to separate out the extensions into piecemeal packages for the testing targets instead.

File: Package.swift [hunk 2]
```diff
@@ -33,6 +37,12 @@ let package = Package(
                 .target(name: "UtilityBeltNetworking"),
             ]
         ),
+        .target(
+            name: "ShamTestHelpers",
```
Comment (bdrelling) [inline Package.swift:41]: I'd prefer we update the new target names to be more explicit about what they are intended for. 

For example, this one could be `ShamXCTestSupport` or something. I'm even fine with introducing a convention like adding an underscore for extension modules, like `Sham_XCTestSupport`, especially given that these are primarily going to be used for incompatibilities for `XCTest` and so on.

File: Package.swift [hunk 3]
```diff
@@ -41,11 +51,18 @@ let package = Package(
                 .target(name: "Sham"),
             ]
         ),
+        .target(
+            name: "UtilityBeltUITestingHelpers",
```
Comment (bdrelling) [inline Package.swift:55]: Same as the previous comment--I don't know what this module is for, by itself, so I wouldn't know when I need to import this.

**Review by bdrelling — CHANGES_REQUESTED** [review]
I did some additional digging after it didn't sit right with both of us but I can't find a cleaner way to handle this. I requested two changes that'll help and then I think we should get this merged in and revisit if we find a better pattern later.

**Review by bdrelling — APPROVED** [review]

**Pull request merged**